akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -81,9 +86,12 @@ class Http2ServerDemux extends GraphStage[BidiShape[Http2SubStream, FrameEvent,
     new GraphStageLogic(shape) with Http2MultiplexerSupport with Http2StreamHandling with GenericOutletSupport with StageLogging {
       logic ⇒
 
+      private val remoteAddressHeader = inheritedAttributes.get[HttpAttributes.RemoteAddress].flatMap(_.address)
```

Comment:
I agree that's one way to do it. It would be nicer if we could create the header in `RequestParsing` directly. Maybe we could add a variant of `def mapStateful(f: Attributes => T => U): Flow[T, U]` that would allow us to extract the attributes directly in `RequestParsing`?